### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by ifielker**
- fixing some lint warnings 
- replacing some chained promises with async/await

### CODE REVIEW

File: src/error.ts [hunk 1]
```diff
@@ -33,6 +33,41 @@ export class FirebaseError extends Error {
   }
 }
 
+/**
+ * Safely gets an error message from an unknown object
+ * @param err an unknown error type
+ * @param defaultMsg an optional message to return if the err is not Error or string
+ * @return An error string
+ */
+export function getErrMsg(err: unknown, defaultMsg?: string): string {
```
Comment (joehan) [inline src/error.ts:42]: Love this -  those `unsafe any access` warnings on every `catch(err: any)` are very obnoxious

File: src/error.ts [hunk 2]
```diff
@@ -33,6 +33,41 @@ export class FirebaseError extends Error {
...
+  }
+  return JSON.stringify(err);
+}
+
+function isObject(value: unknown): value is Record<string, unknown> {
+  return typeof value === "object" && value !== null;
+}
+
+/**
+ * Safely gets a status from an unknown object if
```
Comment (joehan) [inline src/error.ts:58]:
```suggestion
 * Safely gets a status from an unknown object if it has one.
```

File: src/extensions/extensionsApi.ts [hunk 3]
```diff
@@ -458,14 +467,14 @@ export async function getExtensionVersion(extensionVersionRef: string): Promise<
       populateSpec(res.body.spec);
     }
     return res.body;
-  } catch (err: any) {
-    if (err.status === 404) {
+  } catch (err: unknown) {
+    if (isObject(err) && err.status === 404) {
```
Comment (joehan) [inline src/extensions/extensionsApi.ts:471]: Should this just use getErrStatus?

**Review by joehan — APPROVED** [review]

File: src/extensions/extensionsApi.ts (hunk 3 above)
Comment (ifielker) [reply src/extensions/extensionsApi.ts:471]: yup, I wrote this before I wrote that, and forgot to update it.

**Pull request merged**